temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowOutboundCallsInterceptor.java

```diff
@@ -27,16 +27,141 @@
 import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
 import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptorBase;
 import io.temporal.opentracing.OpenTracingOptions;
+import io.temporal.workflow.Functions;
+import io.temporal.workflow.Promise;
 import io.temporal.workflow.Workflow;
 import io.temporal.workflow.WorkflowInfo;
 import io.temporal.workflow.unsafe.WorkflowUnsafe;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 public class OpenTracingWorkflowOutboundCallsInterceptor
     extends WorkflowOutboundCallsInterceptorBase {
   private final SpanFactory spanFactory;
   private final Tracer tracer;
   private final ContextAccessor contextAccessor;
 
+  private class SpanTransferringExecutor implements Executor {
```

Comment:
We can remove this executor and the interceptor for it now since it is not needed

Reply:
Yeah, I was debating that too.  My thought was leaving it would mean any Promise could benefit from the wrapping. Still, my primary concern is supporting propagation to Activity/LocalActivity, so I am happy either way and can remove this.  This decision can always be revisited in a later PR if generalizing propagation to any Promise becomes a desired feature.

Reply:
We can never remove an extension point after we add it. So if it is not needed to solve this problem we shouldn't add it. We can always add it later if the need arises.

Reply:
removed